pairtools/cli/sort.py

```diff
@@ -191,16 +193,47 @@ def sort_py(
             compress_program = "gzip"
 
     column_names = headerops.extract_column_names(header)
-    columns = [c1, c2, p1, p2, pt] + list(extra_col)
-    # Now generating the "-k <i>,<i><mode>" expressions for all columns.
-    # If column name is in the default pairsam format and has an integer dtype there, do numerical sorting
+    column_names = headerops.canonicalize_columns(column_names)
+
+    # Get column indices with fallbacks
+    try:
+        col1 = headerops.get_column_index(column_names, c1)
+        col2 = headerops.get_column_index(column_names, c2)
+        colp1 = headerops.get_column_index(column_names, p1)
+        colp2 = headerops.get_column_index(column_names, p2)
+
+        # Make pair_type optional
+        try:
+            colpt = headerops.get_column_index(column_names, pt) if pt else None
+        except ValueError:
+            colpt = None
+
+        extra_cols = []
+        for col in extra_col:
+            try:
+                extra_cols.append(headerops.get_column_index(column_names, col))
+            except ValueError:
+                logger.warning(f"Extra column {col} not found in header, skipping")
+                continue
+    except ValueError as e:
+        raise ValueError(f"Column error: {str(e)}") from e
+
+    # Generate sort command columns
     cols = []
-    for col in columns:
-        colindex = int(col) if col.isnumeric() else column_names.index(col) + 1
-        cols.append(
-            f"-k {colindex},{colindex}{'n' if issubclass(pairsam_format.DTYPES_PAIRSAM.get(column_names[colindex-1], str), int) else ''}"
-        )
+    sort_columns = [
+        col1,
+        col2,
+        colp1,
+        colp2,
+        colpt,
+    ] + extra_cols  # Order: chrom1, chrom2, pos1, pos2, pair_type
+    for col in sort_columns:
+        if col is None:
+            continue  # Skip optional columns
+        dtype = pairsam_format.DTYPES_PAIRSAM.get(column_names[col], str)
```

Comment:
Looks like the user can define custom columns and their format might be not defined: https://github.com/open2c/pairtools/blob/master/pairtools/lib/pairsam_format.py

It will be great to have a specialized error message